PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs

```diff
@@ -123,15 +133,81 @@ public List<DoctorAppointment> GetAppointmentsForDoctor(int id)
+        }
+
+        public DoctorAppointment RecommendAnAppointment(DoctorUser doctor, DateTime startDate, DateTime endDate, PatientUser patient)
         {
             TimeSpan time1 = TimeSpan.FromMinutes(15);
-            for (var date = date1; date <= date2; date = date.AddDays(1))
+
+            for (var date = startDate; date <= endDate; date = date.AddDays(1))
             {
-                if (getAvailableTerm(doctor, date, time1, patient) != null) return getAvailableTerm(doctor, date, time1, patient);
+                if (GetAvailableTerm(doctor, date, time1, patient) != null) return GetAvailableTerm(doctor, date, time1, patient);
             }
             return null;
         }
+
+        /// <summary> This method is getting List of <c>DoctorAppointment</c> when doctor and patient are available </summary>
+        /// <param name="doctorId"><c>DoctorUser</c> id of doctor who will execute appointment</param>
+        /// <param name="startDate">From which date we want to get recommended appointment</param>
+        /// <param name="endDate">To which date we want to get recommended appointment</param>
+        /// <param name="patientId"><c>PatientUser</c> id of patient that is scheduling appointment</param>
+        /// <returns> List of recommended appointments</returns>
+        public List<DoctorAppointment> GetAllAvailableAppointmentsForRecommendedDates(int doctorId, DateTime startDate, DateTime endDate, int patientId)
+        {
+            List<DoctorAppointment> availableAppointments = new List<DoctorAppointment>();
+            for (var date = startDate; date <= endDate; date = date.AddDays(1))
+            {
+                List<TimeSpan> startTimesAppointments = GetAllStartTimes(CreateAppointmentSetForDate(date, doctorId, patientId).ToList());
+                
+                Shift doctorShift = employeesScheduleService.getShiftForDoctorForSpecificDay(MakeStringFromDate(date), doctorService.GetByid(doctorId));
+                if (doctorShift == null)
+                {
+                    continue;
+                }
+                TimeSpan time = TimeSpan.Parse(doctorShift.startTime);
+                while (time != TimeSpan.Parse(doctorShift.endTime))
+                {
+                    if (!startTimesAppointments.Contains(time) && !IsTermNotAvailable(doctorService.GetByid(doctorId), time, MakeStringFromDate(date), _patientRepository.Find(patientId)))
+                    {
+                        availableAppointments.Add((new DoctorAppointment(0, time, MakeStringFromDate(date), _patientRepository.Find(patientId), doctorService.GetByid(doctorId), new List<Referral>(), doctorService.GetByid(doctorId).ordination)));
+                        if (availableAppointments.Count >= 5)
+                        {
+                            return availableAppointments;
+                        }
+                    }
+                    time = time.Add(TimeSpan.FromMinutes(15));
+                }
+            }
```

Comment:
It is necessary to refactor the function. Separate into several smaller functions and call to the main.